packages/opentelemetry-id-generator-aws-xray/src/platform/node/AWSXRayIdGenerator.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IdGenerator } from '@opentelemetry/core';
+
+const SPAN_ID_BYTES = 8;
+const TRACE_ID_BYTES = 16;
+const EPOCH_BYTES = 4;
+
+/**
+ * IdGenerator that generates trace IDs conforming to AWS X-Ray format.
+ * https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids
+ */
+export class AWSXRayIdGenerator implements IdGenerator {
+  /**
+   * Returns a random 16-byte trace ID formatted/encoded as a 32 lowercase hex
+   * characters corresponding to 128 bits. The first 4 bytes correspond to the current
+   * time, in seconds, as per X-Ray trace ID format.
+   */
+  generateTraceId = (): string => {
+    const epoch = Math.floor(Date.now() / 1000).toString(16);
+    const rand = generateRandomBytes(TRACE_ID_BYTES - EPOCH_BYTES);
+    return epoch + rand;
+  };
+
+  /**
+   * Returns a random 8-byte span ID formatted/encoded as a 16 lowercase hex
+   * characters corresponding to 64 bits.
+   */
+  generateSpanId = (): string => {
+    return generateRandomBytes(SPAN_ID_BYTES);
+  };
+}
+
+const SHARED_BUFFER = Buffer.allocUnsafe(TRACE_ID_BYTES);
+
+function generateRandomBytes(bytes: number): string {
```

Comment:
This seems to be a copy paste from core. We could consider to export it there and reuse it here in a followup PR.